### PR TITLE
add schema migration

### PR DESCRIPTION
**Pull request by siyao-Siyang**
**1. Add SQL Schema migration in Azure SQL Migration extension**
**2. Fix 'save and close' in step 1, 2, 3**
The root cause  is that when resume the previously saved session, calling wizard.setCurrentPage will trigger wizard validator before finishing all of components initialization and thus cause the wizard.setCurrentPage failure. Therefore, when resume the previously session, skip to validate the wizard and use the saved info to initialize the wizard.
**3. Fix the issue that dropdown components cannot be loaded when go back to step 3 after change a target type in step 2**

**The screen shot for schema migration:**
image

**Three scenarios:**
**1. Full schema missing on target**
1.1 select all missing tables (schema + data)
image

1.2 select some of missing tables (schema + data)
image
image

1.3 no tables selection (schema only)
image
image

**2. Partial schema missing on target**
image

**3. Full schema on target**
image
image
image

**4. Failed schema migration**
image

### CODE REVIEW

File: extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts [hunk 1]
```diff
@@ -112,43 +157,144 @@ export class TableMigrationSelectionDialog {
 		}
 	}
 
+	private async _createMigrationContainer(view: azdata.ModelView): Promise<azdata.FlexContainer> {
+		// Schema migration components
+		const schemaMigrationHeader = view.modelBuilder.text()
+			.withProps({
+				value: "Schema migration",
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts:164]: be sure to convert this string to a resource string

File: extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts [hunk 2]
```diff
@@ -112,43 +157,144 @@ export class TableMigrationSelectionDialog {
...
+				requiredIndicator: true,
+				CSSStyles: { ...styles.SECTION_HEADER_CSS, 'margin-top': '4px' }
+			}).component();
+
+		this._schemaMigrationCheckBox = view.modelBuilder.checkBox()
+			.withProps({
+				checked: false,
+				label: "Migrate schema to target",
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts:172]: same here, needs to be a resource string

File: extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts [hunk 3]
```diff
@@ -112,43 +157,144 @@ export class TableMigrationSelectionDialog {
...
+					}
+				}
+			)
+		)
+
+		// Data migration component
+		const dataMigrationHeader = view.modelBuilder.text()
+			.withProps({
+				description: constants.DATA_MIGRATION_INFO,
+				value: "Data migration",
```
Comment (brian-harris) [inline extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts:221]: same here as well, add string to resource string

File: extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts (hunk 1 above)
Comment (siyao-Siyang) [reply extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts:164]: sure. This is props value. so dont need to be resource string. but i will change lable, text etc to resource string

File: extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts (hunk 2 above)
Comment (siyao-Siyang) [reply extensions/sql-migration/src/dialog/tableMigrationSelection/tableMigrationSelectionDialog.ts:172]: done.

**Comment by siyao-Siyang** [comment]
test all scenarios

**Pull request merged**